crates/oxc_linter/src/disable_directives.rs

```diff
@@ -211,10 +264,16 @@ impl DisableDirectives {
                         }
                         match &interval.val {
                             DisabledRule::Single { rule_name, name_span, .. } => {
-                                Some(RuleCommentRule {
-                                    rule_name: rule_name.clone(),
-                                    name_span: *name_span,
-                                })
+                                // Only report known rules as unused
+                                // Unknown rules (e.g., from custom plugins) are silently ignored
+                                if is_known_rule(rule_name) {
```

Comment:
I think we can have better behaviour here.

If the disabled thing is `// oxlint-disable-next-line` (vs `eslint-disable`), we should still report an error.

Since people maybe running `eslint` and `oxlint` in parallel, disable directives won't really work when using `eslint-disable` and oxlint's disable directives.